lib/icingadb/icingadb-objects.cpp

```diff
@@ -2037,7 +2075,10 @@ Dictionary::Ptr IcingaDB::SerializeState(const Checkable::Ptr& checkable)
 
 	tie(host, service) = GetHostService(checkable);
 
-	attrs->Set("id", GetObjectIdentifier(checkable));;
+	String id = GetObjectIdentifier(checkable);
+
+	attrs->Set("id", id);
+	attrs->Set("object_id", id);
```

Comment:
Why is this added with two different keys?

Reply:
To rephrase it (what helped me to understand it):
* `id`: This is the ID of the host/service state object
* `object_id`: This is the foreign key to the object (i.e. host for host states and service for service states) that this object is the state of

As there is a 1:1 relationship between host and host state, the host ID is also used as the host state ID. These are duplicated to 1) avoid having special handling for this in Icinga DB and 2) to have both a primary key and a foreign key in the SQL database in the end.

I think we should add this as a comment. It took me quite some time (and additional discussion) to understand this from the code, even though we talked about this before.

Reply:
Now there's a mix of "host" and "checkable" in the comment. I'd avoid using "checkable" in the comment and just describe it for the host and then end with "This applies to services analogously."

Also I'd rephrase it to better show what happens with the values: "[...], the host ID ('object_id', ends up as a foreign key 'host_state.host_id' referring to 'host.id' in the database) is also used as the host state ID ('id', ends up a the primary key 'host_state.id')"

Reply:
Can’t Icinga DB grab object_id from id?